### DISCUSSION TITLE
Firebird embedded server

### DISCUSSION THREAD

**Discussion by zsolt777** [dotnetcore/FreeSql#798] [Q&A]
Could you tell me please if FreeSql supports Firebird embedded server? 
If so, could you point me in the right direction?

Thank you!

**Comment by 2881099** [comment]
add package
```
dotnet add packages FreeSql
dotnet add packages FreeSql.Provider.Firebird
```

code
```c#
    static Lazy<IFreeSql> firebirdLazy = new Lazy<IFreeSql>(() => new FreeSql.FreeSqlBuilder()
        .UseConnectionString(FreeSql.DataType.Firebird, @"database=localhost:D:\fbdata\EXAMPLES.fdb;user=sysdba;password=123456;max pool size=5")
        .UseAutoSyncStructure(true)
        .UseNameConvert(FreeSql.Internal.NameConvertType.ToUpper)
        .UseNoneCommandParameter(true)

        .UseMonitorCommand(
            cmd => Console.WriteLine(cmd.CommandText),
            (cmd, traceLog) => Console.WriteLine(traceLog))
        .Build());
    public static IFreeSql firebird => firebirdLazy.Value;
```

**Comment by 2881099** [comment]
[freesql_firebird_embed.zip](https://github.com/user-attachments/files/18879676/freesql_firebird_embed.zip)

**Reply by zsolt777** [reply]
Thank you for your efforts!!! Running this sample throws an exception:

15:22:03:507	FreeSql: Master Database数据库出错（执行SQL）〓〓〓〓〓〓〓〓〓〓〓〓〓〓〓
15:22:03:507	INSERT INTO "CustomerMonthBill"("Id", "CompanyId", "BillingMonth", "CustomerId", "TaxAmt", "Amt", "Amt2", "PaidAmt", "VoucherId", "BillState", "CheckOn", "UserId", "RoleId", "CreateOn") SELECT FIRST 1 'c2dc0060-1770-42f9-83da-015c3374ab7c', 'cc31aaba-d24a-4ba5-9f6e-1637ef622439', '2020-07', '9197b718-76aa-41b6-873d-b3dcdbec8faf', 0, 7656.0, 0, 7656.0, '00000000-0000-0000-0000-000000000000', 1, timestamp '2020-08-24 00:00:00.000', '00000000-0000-0000-0000-000000000000', '00000000-0000-0000-0000-000000000000', timestamp '2021-01-22 00:00:00.000' FROM rdb$database 
15:22:03:507	UNION ALL
...
...
15:22:03:507	 SELECT FIRST 1 'a653d1d7-2a11-40f0-990d-c321896f7470', 'cc31aaba-d24a-4ba5-9f6e-1637ef622439', '2020-10', '2756489f-d3c5-4dcc-a4e3-b808f27c7d7c', 0, 40442.0, 0, 40442.0, '00000000-0000-0000-0000-000000000000', 1, timestamp '2020-12-10 00:00:00.000', '00000000-0000-0000-0000-000000000000', '00000000-0000-0000-0000-000000000000', timestamp '2021-01-22 00:00:00.000' FROM rdb$database
15:22:03:507	Dynamic SQL Error
15:22:03:507	SQL error code = -104
15:22:03:507	Unexpected end of command - line 0, column 0
15:22:03:507	线程1事务FreeSql: RollBack

**Reply by 2881099** [reply]
```c#
fsql.Insert(list).BatchOptions(100, 100).ExecuteAffrows();
```

or 

Set ConnectionString PacketSize=32767

However, the maximum value of PacketSize can only be 32767, and the length of the inserted SQL is much longer than this value, so the above scheme still needs to be used.

```c#
database=c:\\firebird_test001.fdb;
client library=fbclient.dll;
user id=SYSDBA;
password=123456;
server type=1;
PacketSize=32767
```
